### PR TITLE
add delete method to lightdb api

### PR DESCRIPTION
**Pull request by vancefarren**

### CODE REVIEW

File: include/net/golioth.h [hunk 1]
```diff
@@ -295,6 +295,21 @@ int golioth_lightdb_set(struct golioth_client *client, const uint8_t *path,
...
+ *
+ * @param client Client instance
+ * @param path Light DB resource path
+ * @param format Format of payload
+ *
+ * @retval 0 On success
+ * @retval <0 On failure
+ */
+int golioth_lightdb_delete(struct golioth_client *client, const uint8_t *path,
+			enum coap_content_format format);
```
Comment (mniestroj) [inline include/net/golioth.h:311]: `format` should not be needed, as there is no payload actually sent.

**Review by mniestroj — CHANGES_REQUESTED** [review]

File: include/net/golioth.h (hunk 1 above)
Comment (vancefarren) [reply include/net/golioth.h:311]: Ah, I thought that might be tied to the path specification in some way.  I'll test without.

File: net/golioth/golioth.c [hunk 2]
```diff
@@ -498,6 +498,32 @@ int golioth_lightdb_set(struct golioth_client *client, const uint8_t *path,
...
+	}
+
+	err = coap_packet_append_option(&packet, COAP_OPTION_URI_PATH,
+					path, strlen(path));
+	if (err) {
+		LOG_ERR("Unable add uri path to packet");
+		return err;
+	}
+
+	return golioth_send_coap_payload(client, &packet, data, data_len);
```
Comment (mniestroj) [inline net/golioth/golioth.c:524]:
```suggestion
	return golioth_send_coap(client, &packet);
```

No reason to send any payload.

Comment (vancefarren) [reply net/golioth/golioth.c:524]: I didn't see that function. Thanks!

**Comment by mniestroj** [comment]
@vancefarren Could you please cleanup that PR and just leave one commit with the required changes? There are also checkpatch and gitlint issues in CI that need to be addressed.

**Comment by mniestroj** [comment]
@vancefarren ping

**Comment by vitprajzler** [comment]
All checks have passed! :partying_face:

**Review by mniestroj — APPROVED** [review]

**Pull request merged**